### PR TITLE
tables: Adding list indexing to darwin plist table

### PR DESCRIPTION
**Pull request by muffins**
This addresses #3544

Some plists might have sublists, and it seemed that while we were processing these correctly it may be the case that the plist subitems don't have a label as it's enumerating list values. This makes an attempt to grab the `Name` field if it exists, and use this as the Label for list objects. This is particularly useful in the case of the MobileMeAccounts plist file.

### CODE REVIEW

**Comment by osqueryer** [comment]
:-1: The commit 354bca4990dcfa5572d27e574253f68ad3308873 (Job results: [1588](https://jenkins.osquery.io/job/osqueryPullRequestOSX/1588/)) failed one or more tests (**macOS/OS X**).

**Comment by theopolis** [comment]
My 2c is maybe this should be a separate table. @groob or @arubdesu should weigh in on pros/cons, will defer to their preference.

**Comment by muffins** [comment]
@theopolis yeah I agree with that. It doesn't make sense changing this to fit one specific use case. Perhaps this PR could be changed to just add a numeric value in the separation? So that it's `Services/0/status` as opposed to `Services//status`? Then we could just roll another vtable getting our specific use case addressed.

**Comment by theopolis** [comment]
Sounds great!

**Comment by muffins** [comment]
Here's what new queries look like:
```
osquery> select key, subkey, value from plist where path like '/Users/%/Library/Preferences/MobileMeAccounts.plist';
+----------+-------------------------------+------------------------------------------------+
| key      | subkey                        | value                                          |
+----------+-------------------------------+------------------------------------------------+
| Accounts | DisplayName                   | Nick Anderson                                  |
| Accounts | lastName                      | Anderson                                       |
| Accounts | firstName                     | Nick                                           |
| Accounts | isManagedAppleID              | 0                                              |
| Accounts | beta                          | 0                                              |
| Accounts | LoggedIn                      | 1                                              |
| Accounts | Services/0/status             | active                                         |
| Accounts | Services/0/Name               | CLOUDDESKTOP                                   |
| Accounts | Services/0/ServiceID          | com.apple.Dataclass.CloudDesktop               |
| Accounts | Services/1/ServiceID          | com.apple.Dataclass.Family                     |
| Accounts | Services/1/Name               | FAMILY                                         |
| Accounts | Services/1/showManageFamily   | 1                                              |
| Accounts | Services/2/apsEnv             | production                                     |
| Accounts | Services/2/url                | https://p34-ubiquity.icloud.com:443            |
| Accounts | Services/2/authMechanism      | token                                          |
| Accounts | Services/2/Enabled            | 0                                              |
| Accounts | Services/2/Name               | MOBILE_DOCUMENTS                               |
| Accounts | Services/2/wsUrl              | https://p34-ubiquityws.icloud.com:443          |
| Accounts | Services/2/ServiceID          | com.apple.Dataclass.Ubiquity                   |
```

**Comment by osqueryer** [comment]
:-1: The commit cc52ab6e48c9e2704ceb2eac42c3c8247184e5e7 (Job results: [1674](https://jenkins.osquery.io/job/osqueryPullRequestOSX/1674/)) failed one or more tests (**macOS/OS X**).

File: osquery/tables/system/darwin/preferences.cpp [hunk 1]
```diff
@@ -244,9 +245,14 @@ void genOSXPlistPrefValue(const pt::ptree& tree,
     Row r = base;
     if (r["subkey"].size() > 0) {
       r["subkey"] += "/";
+      if (item.first.size() == 0) {
+        r["subkey"] += std::to_string(level);
```
Comment (theopolis) [inline osquery/tables/system/darwin/preferences.cpp]: You could write `r["subkey"] += std::to_string(level++);`

File: osquery/tables/system/darwin/preferences.cpp [hunk 2]
```diff
@@ -244,9 +245,14 @@ void genOSXPlistPrefValue(const pt::ptree& tree,
...
+        level++;
```
Comment (theopolis) [inline osquery/tables/system/darwin/preferences.cpp]: and nix this.

**Pull request merged**